src/core/UserIdentity.cpp

```diff
@@ -152,3 +178,75 @@ bool UserIdentity::isServicePublished() const
     return m_hiddenService && m_hiddenService->status() >= Tor::HiddenService::Published;
 }
 
+#ifdef PROTOCOL_NEW
+/* Handle an incoming connection to this service
+ *
+ * A Protocol::Connection is created to handle this socket. The
+ * connection initially has a purpose of Unknown. It times out
+ * and automatically closes after ConnectionPrivate::UnknownPurposeTimeout
+ * seconds, unless the purpose is changed.
+ *
+ * If the connection successfully completes authentication,
+ * handleIncomingAuthedConnection is called to link it to a ContactUser
+ * (if applicable) and set the purpose.
+ */
+void UserIdentity::onIncomingConnection()
+{
+    while (m_incomingServer->hasPendingConnections()) {
+        QTcpSocket *socket = m_incomingServer->nextPendingConnection();
+        socket->setProperty("localHostname", m_hiddenService->hostname());
```

Comment:
It is used to implement `Connection::serverHostname`. It could be done more clearly by adding a `Connection::setServerHostname` method. Do you think that is necessary?